pkg/gvks/supportedgvks/supportedgvks.go

```diff
@@ -42,6 +67,43 @@ func resources(smLoader *servicemappingloader.ServiceMappingLoader, serviceMetaL
 	return gvks
 }
 
+func DirectResources() (map[schema.GroupVersionKind]bool, error) {
+	crds, err := crdloader.LoadCRDs()
+	if err != nil {
+		return nil, fmt.Errorf("error loading crds: %w", err)
+	}
+	handWrittenIAMTypes := make(map[schema.GroupVersionKind]bool)
+	directResources := make(map[schema.GroupVersionKind]bool)
+	for _, gvk := range BasedOnHandwrittenIAMTypes() {
+		handWrittenIAMTypes[gvk] = true
+	}
+	for _, crd := range crds {
+		if crd.ObjectMeta.Labels["cnrm.cloud.google.com/tf2crd"] == "true" {
```

Comment:
For SQLInstance, we have "partially enabled" the direct controller. This means:
- The tf2crd label is present in the CRD
- Only when spec.cloneSource is specified, the direct controller will manage the resource. Otherwise, the tf controller will manage the resource.

Therefore, this function will not report SQLInstance as a direct resource (it will think it's a terraform-based resource). Do you anticipate this will cause any issues?

Reply:
We only want to identify pure direct resources using this functionm, so the mixed resource won't be causing any issues.

Reply:
Cool, thanks @maqiuyujoyce !